projects/storefrontlib/src/layout/direction/direction.service.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  ConfigInitializerService,
+  LanguageService,
+  WindowRef,
+} from '@spartacus/core';
+import { EMPTY, of, Subject } from 'rxjs';
+import { DirectionMode } from '../config/direction.model';
+import { DirectionService } from './direction.service';
+
+class MockWindowRef implements Partial<WindowRef> {
+  document = { documentElement: { nodeName: 'document' } } as Document;
+}
+
+class MockLanguageService implements Partial<LanguageService> {
+  getActive = () => EMPTY;
+}
+
+class MockConfigInitializerService {
+  getStableConfig() {}
+}
+
+describe('DirectionService', () => {
+  let service: DirectionService;
+  let languageService: LanguageService;
+  let winRef: WindowRef;
+  let configInitializerService: ConfigInitializerService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: WindowRef, useClass: MockWindowRef },
+        { provide: LanguageService, useClass: MockLanguageService },
+        {
+          provide: ConfigInitializerService,
+          useClass: MockConfigInitializerService,
+        },
+      ],
+    });
+    service = TestBed.inject(DirectionService);
+    languageService = TestBed.inject(LanguageService);
+    winRef = TestBed.inject(WindowRef);
+    configInitializerService = TestBed.inject(ConfigInitializerService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('addDirection', () => {
+    it('should add dir=ltr attribute to element', () => {
+      const el = {} as HTMLElement;
+      service.addDirection(el, DirectionMode.LTR);
+      expect(el.dir).toBe(DirectionMode.LTR);
+    });
+
+    it('should add dir=rtl attribute to element', () => {
+      const el = {} as HTMLElement;
+      service.addDirection(el, DirectionMode.RTL);
+      expect(el.dir).toBe(DirectionMode.RTL);
+    });
+  });
+
+  describe('getDirection', () => {
+    describe('without default', () => {
+      beforeEach(() => {
+        spyOn(configInitializerService, 'getStableConfig').and.returnValue(
+          of({
+            direction: {
+              detect: true,
+              ltrLanguages: ['en', 'de'],
+              rtlLanguages: ['ar', 'he'],
+            },
+          }).toPromise()
```

Comment:
Bagatelle: `of(value).toPromise()` can be replaced with simply `Promise.resolve(value)`